lib/core/src/sdk.rs

```diff
@@ -1132,48 +1165,146 @@ impl LiquidSdk {
+            PayOnchainAmount::Drain => drain_receiver_amount_sat,
+        };
+        let is_drain = match req.amount {
+            PayOnchainAmount::Receiver { amount_sat } => amount_sat == drain_receiver_amount_sat,
+            PayOnchainAmount::Drain => true,
+        };
+        let (payer_amount_sat, total_fees_sat) = match is_drain {
+            true => {
+                self.validate_user_lockup_amount_for_chain_pair(
+                    &pair,
+                    drain_user_lockup_amount_sat,
+                )?;
+
+                (drain_payer_amount_sat, drain_total_fees_sat)
+            }
+            false => {
+                let user_lockup_amount_sat_without_service_fee =
+                    receiver_amount_sat + claim_fees_sat + server_fees_sat;
+
+                // The resulting invoice amount contains the service fee, which is rounded up with ceil()
+                // Therefore, when calculating the user_lockup amount, we must also round it up with ceil()
+                let user_lockup_amount_sat = (user_lockup_amount_sat_without_service_fee as f64
+                    * 100.0
+                    / (100.0 - pair.fees.percentage))
+                    .ceil() as u64;
+                self.validate_user_lockup_amount_for_chain_pair(&pair, user_lockup_amount_sat)?;
+
+                let lockup_fees_sat = self
+                    .estimate_lockup_tx_fee_chain_send(user_lockup_amount_sat)
+                    .await?;
+
+                let boltz_fees_sat =
+                    user_lockup_amount_sat - user_lockup_amount_sat_without_service_fee;
+                let total_fees_sat =
+                    boltz_fees_sat + lockup_fees_sat + claim_fees_sat + server_fees_sat;
+                let payer_amount_sat = receiver_amount_sat + total_fees_sat;
+
+                (payer_amount_sat, total_fees_sat)
+            }
+        };
 
         let res = PreparePayOnchainResponse {
             receiver_amount_sat,
             claim_fees_sat,
-            total_fees_sat: boltz_fees_sat + lockup_fees_sat + claim_fees_sat + server_fees_sat,
+            total_fees_sat,
         };
 
-        let payer_amount_sat = res.receiver_amount_sat + res.total_fees_sat;
         ensure_sdk!(
-            payer_amount_sat <= self.get_info().await?.balance_sat,
+            payer_amount_sat <= balance_sat,
             PaymentError::InsufficientFunds
         );
 
         Ok(res)
     }
 
+    /// Find the maximum `receiver_amount_sat` that can be used as argument in `prepare_pay_onchain`.
+    ///
+    /// Note that when using this amount in `prepare_pay_onchain`, the resulting `pay_onchain` will
+    /// leave 1 sat in the wallet. To send all funds and leave no sats in the wallet,
+    /// `prepare_pay_onchain` should be called with `PayOnchainAmount::Drain`.
+    ///
+    /// ### Arguments
+    /// - `sat_per_vbyte`: optional user-chosen BTC claim tx feerate
+    pub async fn get_max_receiver_amount_sat(
```

Comment:
I don't understand why when we use lwk drain option we get higher maximum than we specify explicitly.
The drain as far as I understand uses all utxos so the tx size should be the largest possible and more than any other tx created in a different way. Am I missing something?

Reply:
The reason for that difference is that the drain option doesn't expect a change output, whereas for a normal send LWK includes a change output. So the tx size is higher for normal send, which affects the fee, which affects the max amount sendable.

Reply:
If that's the case then I think it is normal and we shouldn't do anything about it.
The fact that lwk wallet added a drain flag that behaves differently indicates it is a different cast.
If the user passes drain then the prepare will return X amount.
The user won't be able to send that X amount without the drain and will get insufficient funds error which seems OK to me.

Reply:
As discussed, I reverted the previous commit.

Now we're not exposing a max for the custom amount, but the user can send the full balance with `drain`.